### DISCUSSION TITLE
Lesson 6: Unable to deploy "deploy.py"

### DISCUSSION THREAD

**Discussion by icalaba** [smartcontractkit/full-blockchain-solidity-course-py#633] [Q&A] [answered]
when i deploy the contract i get this ::

Error: Non-hexadecimal digit found
please click on the link to view my screenshot:
[](https://snipboard.io/U07YSl.jpg)

**Comment by cromewar** [comment]
hello @icalaba could you please share more info about the problem? (your code, the timestamp of the video and etc). Also, there is not any link for a screenshot but you can paste directly screenshots here on github.

**Reply by icalaba** [reply]
Hello @cromewar , from the Timestamp between 5:07 to about 5:10 , when i deploy the contract, i am unable to do so , and i get the error message "Error: Non-hexadecimal digit found"
![Screenshot (10)](https://user-images.githubusercontent.com/73090390/147469533-e686b6ec-3678-4106-9d56-2748f1adcb35.png)

**Reply by icalaba** [reply]
Hello @cromewar , from the Timestamp between 5:07 to about 5:10 , when i deploy the contract, i am unable to do so , and i get the error message "Error: Non-hexadecimal digit found"
![Screenshot (10)](https://user-images.githubusercontent.com/73090390/147469533-e686b6ec-3678-4106-9d56-2748f1adcb35.png)

**Reply by cromewar** [reply]
Please delete the build folder, and compile again `brownie compile` does brownie compile correctly the contract?

**Reply by icalaba** [reply]
i deleted the build folder and compiled , and yes the contract is compiled correctly , but issue still persists 
![Screenshot (12)](https://user-images.githubusercontent.com/73090390/147507703-b0765f61-a6a9-42eb-9e2c-ee9be0009728.png)

**Comment by cromewar** [comment, marked as answer]
Does [this](https://ethereum.stackexchange.com/questions/98811/error-non-hexadecimal-digit-found) work for you?

**Reply by icalaba** [reply]
unfortunately no, it returns the same issue (" "Error: Non-hexadecimal digit found")i had earlier reported when i ran the "brownie run scripts/deploy.py --network rinkeby" command

**Reply by cromewar** [reply]
according to the thread on the link, you should add the private key manually, if nothing works this could be a problem with your Python setup so you could try using a python virtualenv or also using WSL.

**Reply by icalaba** [reply]
ok thanks i used WSL and it is working fine , however , this is another issue by the way , when i deploy to local ganache with brownie , i get this message "RPCConnectionError: Able to launch RPC client, but unable to connect." this is from 5:32:53 in the video 
![Screenshot (13)](https://user-images.githubusercontent.com/73090390/147772635-704ab900-e924-45ae-9159-2703d132f2a1.png)

**Reply by cromewar** [reply]
Hello @icalaba soory for the delay, well you have to take in mind WSL is a hole new operating system, so the node packages you have installed on windows won't work here, you should install them over again, I strongly suggest you to check the course index "windows support" on which you'll find a guide about how to configure windows and WSL for smart contract development.

Best regards, Cromewar.... and please let me know if this works for you.

**Reply by icalaba** [reply]
Thank you very much @cromewar , IT WORKED!

**Reply by icalaba** [reply]
i think the issue was WSL had no connection to my windows installed ganache, hence the error, however when i initialized ganache through the wsl terminal using "ganache-cli" then deployed the contract in vscode. it worked. any further suggestions to make wsl work with the windows locally installed ganache UI ?

**Reply by cromewar** [reply]
Yes on ganache UI, go to settings and change the host to the one with WSL on its name.

**Reply by icalaba** [reply]
Hello , I tried that but got this error::
 "ConnectTimeout: HTTPConnectionPool(host='172.21.64.1', port=8545): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7f2ad346e250>, 'Connection to 172.21.64.1 timed out. (connect timeout=30)'))"
![Screenshot (14)](https://user-images.githubusercontent.com/73090390/147845123-80c1ef77-d813-4e29-9862-69fb21ec60ea.png)

**Reply by icalaba** [reply]
however , when i initialized "ganache-cli" on the ubuntu terminal , it worked , this is definitely some issue with the ganache UI , offscript how can i get some test eth for rinkeby, faucet.rinkeby.io seems to not be working. any other alternatives ?

**Reply by cromewar** [reply]
The host should be: http://172.21.64.1, otherwise it won't work.

**Reply by icalaba** [reply]
It is and it is not working either

**Reply by cromewar** [reply]
Could you please share screenshots of your ganache ui, and you code please.

**Comment by icalaba** [comment]
![Screenshot (15)](https://user-images.githubusercontent.com/73090390/147875260-e8e5a6ae-58e4-4ca0-a79b-23d7dfad0a8e.png)

i copied and pasted everything from the github page. so the code is the same

**Reply by icalaba** [reply]
i am currently at 5:53 just so you know what code i am working with currently , and another crazy thing happening is when i try to run "brownie run scripts/deploy.py --network mainnet-fork" i get this :: 
![Screenshot (16)](https://user-images.githubusercontent.com/73090390/147875572-c9025d78-4d09-4471-8176-4f4deef67a9e.png)

**Reply by icalaba** [reply]
as you can see there i was tyring to connect to the mainnet-fork , why is that launching ganache

**Reply by cromewar** [reply]
Oh I see now, look your mainnet-for uses a pre configured setting, if you remember you added that network manually with `brownie networks add` I pretty much sure the host name you set up was different as the one you have now because you changed the host to WSL, so first of all try deleting that network and add it again using the current configuration.

Also make sure your ganache is not blocked by your windows firewall (some windows users have reported this).
 I hope this info might help.
